### ISSUE TITLE
[Feature]: Inline functions on hot path

### ISSUE THREAD

**Issue by rvolosatovs** [enarx/enarx#1731]

**Comment by moksh-pathak** [comment]
I would like to take this up.

**Comment by dpal** [comment]
This is about the rust performance.

**Comment by Ankita13-code** [comment]
Hi @rjzak @mayankkumar2! Can you please tell which crates or specific files are exactly being referred to here? This issue seems about identifying the inline functions and adding the #inline tag right?

**Comment by mayankkumar2** [comment]
@Ankita13-code You are right. But, for the exact create being referenced here @rjzak or someone from the enarx team could answer that.

**Comment by rjzak** [comment]
You need to use a profiling tool to identify the "hot" functions, and see if inlining them makes the code faster.

Fortunately, this PR https://github.com/enarx/enarx/pull/2391 added flame graph support, which indicates which functions are taking the most time. Just because a function takes the lot of time doesn't necessarily mean that inlining will make Enarx faster, but it could offer a place to start. Generally, functions are inlined when they're small (just a few lines) and used a lot. Too many inlines, and the resulting binary might get unnecessarily large.
